src/resources/adc_esp32.cc

```diff
@@ -63,20 +65,20 @@ static adc_atten_t get_atten(int mv) {
   return ADC_ATTEN_DB_11;
 }
 
-struct ADCState {
-  ADCState(adc_unit_t unit, int chan)
-    : unit(unit)
-    , chan(chan) {}
+AdcState::AdcState(SimpleResourceGroup* group) : SimpleResource(group) {
+  // what to do with this?
+}
```

Comment:
Unless I'm missing something you can combine the group, the unit and the chan into one constructor with three arguments.  This is preferred, because then you can't forget to call init.